src/newgrf_debug_gui.cpp

```diff
@@ -402,8 +402,8 @@ struct NewGRFInspectWindow : Window {
 		vseprintf(buf, lastof(buf), format, va);
 		va_end(va);
 
+		if (offset + this->vscroll->GetPosition() >= this->vscroll->GetCapacity()) return;
```

Comment:
Are you sure this is correct?

In the old code, `offset` seems to go from 0..N, with `this->vscroll` could have a Position of M, and a Capacity of X. (where M and X have no relation to each other).

It returns if `offset - M < 0` or `offset - M >= X`. In other words, if `offset < M` or `offset >= X + M`. Basically, if the offset is within the range of the scrollbar.

Now in the new code you do the upper-bound check, but I do not see the lower-bound check. But the statement that is here, seems to be wrong (`offset + M >= X`). It mostly triggered me as I couldn't make sense of this if-statement ;) To me it reads the old code does: if not `M <= offset < M + X`, return. Or in real C++ code:

```
/* Ensure the offset is within the vscroll range */
if (offset < this->vscroll->GetPosition()) retun;
if (offset >= this->vscroll->GetPosition() + this->vscroll->GetCapacity()) return;
```

I might be completely wrong, so please do your own analysis again too, but your new line of code simply does not make sense to me: `offset + M >= X` .. like .. if you scroll for X, this is always true. So nothing renders anymore?